### ISSUE TITLE
Random build failures

### ISSUE THREAD

**Issue by Dkendal** [StemboltHQ/battle_snake#13]
Lots of these, mostly around game servers not existing any more in tests or mnesia records not existing when the test runs.

**Comment by Dkendal** [comment]
```elixir
  1) test integeration tests the game can be resume play (BattleSnake.GameServerTest)
     test/battle_snake/game_server_test.exs:42
     ** (exit) exited in: GenServer.call(#PID<0.1150.0>, :resume, 5000)
         ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
     stacktrace:
       (elixir) lib/gen_server.ex:737: GenServer.call/3
       test/battle_snake/game_server_test.exs:43: (test)
```

**Comment by Dkendal** [comment]
``` elixir  
1) test GameServer.Supervisor.start_game_server/1 starts a game server process (BattleSnake.GameServer.SupervisorTest)
     test/battle_snake/game_server/supervisor_test.exs:18
     match (=) failed
     code:  %{active: 1, workers: 1} = Supervisor.count_children(@sup_name)
     right: %{active: 4, specs: 1, supervisors: 0, workers: 4}
     stacktrace:
       test/battle_snake/game_server/supervisor_test.exs:21: (test)
```

**Comment by Dkendal** [comment]
resolved in #29
